sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileConstants.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file;
+
+/**
+ * Holds the Constants used for the File Service.
+ */
+public final class FileConstants {
+    /**
+     * Default header value for file permission.
+     */
+    public static final String FILE_PERMISSION_INHERIT = "Inherit";
+
+    /**
+     * Default file attribute value for files.
+     */
+    public static final String FILE_ATTRIBUTES_NONE = "None";
+
+    /**
+     * Default file creation and file last write time.
+     */
+    public static final String FILE_TIME_NOW = "Now";
+
+    /**
+     * Default value for several SMB file headers.
+     */
+    public static final String PRESERVE = "Preserve";
+
+    /**
+     * Stores a reference to the date/time pattern expected for File SMB properties
+     */
+    public static final String SMB_DATE_STRING = "yyyy-MM-dd'T'HH:mm:ss.SSSSSSS'Z'";
```

Comment:
Is it required that Files use a more precise date string? Could this use `Utility.ISO8601_PATTERN`? They're both ISO8601 but the one in utility is precise down to the second.

Reply:
The service call will fail with a less precise date

Reply:
I'm pretty sure the service expects this format exactly, down to the millionth place.